src/zenml/utils/tag_utils.py

```diff
@@ -0,0 +1,79 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utility functions for handling tags."""
+
+from typing import List
+from uuid import UUID
+
+from zenml.enums import TaggableResourceTypes
+from zenml.exceptions import EntityExistsError
+from zenml.models.tag_models import TagRequestModel, TagResourceRequestModel
+from zenml.utils.uuid_utils import generate_uuid_from_string
+
+
+def _get_tag_resource_id(tag_id: UUID, resource_id: UUID) -> UUID:
+    return generate_uuid_from_string(str(tag_id) + str(resource_id))
+
+
+def create_links(
+    tag_names: List[str],
+    resource_id: UUID,
+    resource_type: TaggableResourceTypes,
+) -> None:
+    """Creates a tag<>resource link if not present.
+
+    Args:
+        tag_names: The list of names of the tags.
+        resource_id: The id of the resource.
+        resource_type: The type of the resource to create link with
+    """
+    from zenml.client import Client
+
+    zs = Client().zen_store
+    for tag_name in tag_names:
+        try:
+            tag = zs.get_tag(tag_name)
+        except KeyError:
+            tag = zs.create_tag(TagRequestModel(name=tag_name))
+        try:
+            zs.create_tag_resource(
+                TagResourceRequestModel(
+                    tag_id=tag.id,
+                    resource_id=resource_id,
+                    resource_type=resource_type,
+                )
+            )
+        except EntityExistsError:
+            pass
+
+
+def delete_links(
```

Comment:
No idea where to write this - but do we have any way of implementing cascading delete? Like if I delete resource 1 - will all entries for tags on resource 1 be deleted

Reply:
It will be cascaded - this is need for removal of tags via update, I'll rename it for clarity

Reply:
This folk is cascading (set in Models and other taggable entities going forward)
```python
tags: List["TagResourceSchema"] = Relationship(
        back_populates="model",
        sa_relationship_kwargs=dict(
            primaryjoin=f"and_(TagResourceSchema.resource_type=={TaggableResourceTypes.MODEL.value}, foreign(TagResourceSchema.resource_id)==ModelSchema.id)",
            cascade="delete",
        ),
    )
```